src/needless_bool.rs

```diff
@@ -27,20 +27,30 @@ impl LintPass for NeedlessBool {
     }
 
     fn check_expr(&mut self, cx: &Context, e: &Expr) {
-        if let ExprIf(_, ref then_block, Option::Some(ref else_expr)) = e.node {
+        if let ExprIf(ref pred, ref then_block, Some(ref else_expr)) = e.node {
             match (fetch_bool_block(then_block), fetch_bool_expr(else_expr)) {
-                (Option::Some(true), Option::Some(true)) => {
+                (Some(true), Some(true)) => {
                     span_lint(cx, NEEDLESS_BOOL, e.span,
-                              "your if-then-else expression will always return true"); },
-                (Option::Some(true), Option::Some(false)) => {
+                              "this if-then-else expression will always return true"); },
+                (Some(false), Some(false)) => {
                     span_lint(cx, NEEDLESS_BOOL, e.span,
-                              "you can reduce your if statement to its predicate"); },
-                (Option::Some(false), Option::Some(true)) => {
-                    span_lint(cx, NEEDLESS_BOOL, e.span,
-                              "you can reduce your if statement to `!` + its predicate"); },
-                (Option::Some(false), Option::Some(false)) => {
-                    span_lint(cx, NEEDLESS_BOOL, e.span,
-                              "your if-then-else expression will always return false"); },
+                              "this if-then-else expression will always return false"); },
+                (Some(true), Some(false)) => {
+                    let pred_snip = snippet(cx, pred.span, "..");
```

Comment:
in that case it would be `reduce ... to just 'its predicate'` (with backticks instead of quotes), which looks like code.

Might work if we use something more obvious, like `"<predicate>"`.